src/utils.coffee

```diff
@@ -18,7 +18,7 @@ device = require './device'
 exports.supervisorVersion = require('./lib/supervisor-version')
 
 configJson = require('/boot/config.json')
-if Boolean(configJson.supervisorOfflineMode)
+if Boolean(config.apiEndpoint) and !Boolean(configJson.supervisorOfflineMode)
```

Comment:
Ok, because this is just asking for someone to mess up the check in the future, it really would be nice to have consolidation so the complexity can be kept in one place rather than spread around like this